src/core/hle/service/set/set.cpp

```diff
@@ -31,6 +32,44 @@ constexpr std::array<LanguageCode, 17> available_language_codes = {{
     LanguageCode::ZH_HANT,
 }};
 
+enum class KeyboardLayout : u64 {
+    Japanese = 0,
+    EnglishUs = 1,
+    EnglishUsInternational = 2,
+    EnglishUk = 3,
+    French = 4,
+    FrenchCa = 5,
+    Spanish = 6,
+    SpanishLatin = 7,
+    German = 8,
+    Italian = 9,
+    Portuguese = 10,
+    Russian = 11,
+    Korean = 12,
+    ChineseSimplified = 13,
+    ChineseTraditional = 14,
+};
+
+static std::unordered_map<LanguageCode, KeyboardLayout> language_to_layout{{
```

Comment:
```suggestion
static constexpr std::array<std::pair<LanguageCode, KeyboardLayout>, 17> language_to_layout{{
```

and do the lookup with `std::find_if`. This way there's no runtime initializer used at runtime when the program starts, and no heap is used up for the lifetime of the program.